dask/bag/core.py

```diff
@@ -183,6 +183,42 @@ def finalize_item(results):
     return results[0]
 
 
+def merge_kwargs(dsk, kwargs):
+    """ Merges dask values from kwargs with dsk.
+
+    Returns the merged dsk and a graph expression useful for use with apply in
+    a dask graph to pass the final computed **kwargs to a function.
+
+    Currently only dask.bag.Item and python literal values are supported.
+    """
+    kw_pairs = []
+    for key, val in iteritems(kwargs):
+        if isinstance(val, Item):
+            dsk = merge(dsk, val.dask)
+            val = val.key
+        # TODO elif isinstance(val, Value):
+        elif isinstance(val, Base):
+            raise NotImplementedError(
+                '%s not supported as kwarg value to Bag.map_partitions'
+                % type(val).__name__)
+        kw_pairs.append([key, val])
+    return dsk, kw_pairs
```

Comment:
Maybe?  What is the smallest possible change we can make to to_task_dasks
to achieve the desired effect?

On Tue, Mar 15, 2016 at 2:52 PM, Joshua T Corbin <notifications@github.com>
wrote:

> In dask/bag/core.py
> <https://github.com/dask/dask/pull/1054#discussion_r56247599>:
>
> > +    a dask graph to pass the final computed **kwargs to a function.
> > +
> > +    Currently only dask.bag.Item and python literal values are supported.
> > +    """
> > +    kw_pairs = []
> > +    for key, val in iteritems(kwargs):
> > +        if isinstance(val, Item):
> > +            dsk = merge(dsk, val.dask)
> > +            val = val.key
> > +        # TODO elif isinstance(val, Value):
> > +        elif isinstance(val, Base):
> > +            raise NotImplementedError(
> > +                '%s not supported as kwarg value to Bag.map_partitions'
> > +                % type(val).__name__)
> > +        kw_pairs.append([key, val])
> > +    return dsk, kw_pairs
>
> Aye, shall I salvage as mentioned in #1041 (comment)
> <https://github.com/dask/dask/pull/1041#issuecomment-197038147> first? I
> was perhaps over recalibrating here since I am indeed prone to over solving
> things ;-)
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/dask/dask/pull/1054/files/99fc7e7df80eb1def4471dda1611e5e7a577e248#r56247599>
>